src/alerts/Alerts.test.ts

```diff
@@ -0,0 +1,81 @@
+import { fireEvent, screen, within } from '@testing-library/react';
+import { h } from 'react-hyperscript-helpers';
+import * as Utils from 'src/libs/utils';
+import { asMockedFn, renderWithAppContexts as render } from 'src/testing/test-utils';
+
+import { Alert as AlertType } from './Alert';
+import { AlertsIndicator } from './Alerts';
+import { useServiceAlerts } from './service-alerts';
+
+type ServiceAlertsExports = typeof import('./service-alerts');
+jest.mock('./service-alerts', (): ServiceAlertsExports => {
+  const originalModule = jest.requireActual<ServiceAlertsExports>('./service-alerts');
+  return {
+    ...originalModule,
+    useServiceAlerts: jest.fn(),
+  };
+});
+
+const testAlerts: AlertType[] = [
+  {
+    id: 'abc',
+    title: 'The systems are down!',
+    message: 'Something is terribly wrong',
+  },
+  {
+    id: 'def',
+    title: 'Scheduled maintenance',
+    message: 'Offline tomorrow',
+  },
+];
+
+describe('Alerts', () => {
+  beforeEach(() => {
+    asMockedFn(useServiceAlerts).mockReturnValue(testAlerts);
+  });
+
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  it('renders number of alerts', () => {
+    render(h(AlertsIndicator));
+    expect(screen.getByRole('button')).toHaveTextContent(`${testAlerts.length}`);
+  });
+
+  it('renders popup with alerts', () => {
+    render(h(AlertsIndicator));
```

Comment:
// Arrange, // Act, // Assert section comments on all these tests please.

Reply:
Done in https://github.com/DataBiosphere/terra-ui/pull/4400/commits/42f472fa02230a81fac9915b1e827b25e2f85dfe.